Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py

```diff
@@ -62,18 +62,21 @@ def editable(cls, branch, repository=None):
         if not repository or not isinstance(repository, local.Git):
             return False
 
-        # FIXME: Need to consider alternate remotes
-        for remote in ['origin']:
-            if branch in repository.branches_for(remote=remote, cached=True):
-                return False
+        remote_repo = repository.remote(name=repository.default_remote)
+        if remote_repo and isinstance(remote_repo, remote.GitHub):
+            for name in repository.source_remotes():
+                if branch in repository.branches_for(remote=name, cached=True):
+                    return False
         return True
 
     @classmethod
-    def branch_point(cls, repository):
+    def branch_point(cls, repository, limit=None):
         cnt = 0
         commit = repository.commit(include_log=False, include_identifier=False)
         while cls.editable(commit.branch, repository=repository):
             cnt += 1
+            if limit and cnt > limit:
```

Comment:
why do we need a limit? expensive to run? can get too deeply nested?

Reply:
We want a limit because it's possible for folks to have poorly configured checkouts and for this function to find hundreds or thousands of commits. If we've gone past about a dozen, something has probably gone wrong.